clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProviderBuilder.java

```diff
@@ -163,8 +216,29 @@ private void validate() {
         throw new IllegalArgumentException(
             "Expected specified credentials cache to be a file but found directory instead.");
       }
+      if (connectTimeout.isZero() || connectTimeout.isNegative()) {
+        throw new IllegalArgumentException("Expected connectTimeout to be is more that 0.");
+      }
+      validateTimeout(connectTimeout, "ConnectTimeout");
+      validateTimeout(readTimeout, "ReadTimeout");
     } catch (final NullPointerException | IOException e) {
       throw new IllegalArgumentException(e);
     }
   }
+
+  private void validateTimeout(final Duration timeout, final String timeoutName) {
+    final boolean valid;
+    if (timeout.isZero() || timeout.isNegative()) {
+      valid = false;
+    } else {
+      final long millis = timeout.toMillis();
+      valid = millis < Integer.MAX_VALUE && millis > Integer.MIN_VALUE;
+    }
+    if (!valid) {
```

Comment:
```suggestion
    if (timeout.isZero() || timeout.isNegative() || timeout.toMillis() > Integer.MAX_VALUE) {
```
The `> Integer.MIN_VALUE` is redundant at this point so we can check this more a bit more concisely

Reply:
Done